lib/web_ui/test/engine/text_editing_test.dart

```diff
@@ -2423,6 +2465,173 @@ Future<void> testMain() async {
       expect(event.defaultPrevented, isFalse);
     });
 
+    test('inserts element in the correct view', () {
+      final DomElement host = createDomElement('div');
+      domDocument.body!.append(host);
+      final EngineFlutterView view = EngineFlutterView(dispatcher, host);
+      dispatcher.viewManager.registerView(view);
+
+      textEditing = HybridTextEditing();
+      showKeyboard(inputType: 'text', viewId: view.viewId);
+
+      final DomElement input = textEditing!.strategy.domElement!;
+
+      // Input is appended to the right view.
+      expect(view.dom.textEditingHost.contains(input), isTrue);
```

Comment:
(I think this is the line that safari doesn't like?)

One way of making the text more readable is to add a `reason: 'input must be contained in textEditingHost'` or similar, so the test failure has some meaning (rather than: "expected true but got false")

Reply:
Done. This is because on Safari we do some shady stuff haha. I updated the test to properly work on Safari.